### DISCUSSION TITLE
Migrate non-HA cluster to HA?

### DISCUSSION THREAD

**Discussion by kralicky** [k3s-io/k3s#2993] [Q&A]
Hi, I am currently operating a k3s cluster with the kine/sqlite backend and a single control-plane node. I am wondering if it is possible to upgrade this cluster to use HA mode with embedded etcd without needing to start from scratch. If so, what would be the proper procedure to do this? Can "etcd" snapshots be taken from kine and restored into a real etcd?
Thanks!

**Comment by brandond** [comment]
There is not currently any tool to directly migrate between backends.

**Reply by timoschwarzer** [reply]
Dang, I recently stumbled upon @ibuildthecloud 's [tweet](https://twitter.com/ibuildthecloud/status/1257748334011224064) and thought I just missed something... :sweat_smile: 
Is upgrading sqlite to etcd still a planned feature?

**Reply by brandond** [reply]
I'm not sure what Darren was referring to there 🤷🏻

**Reply by warmchang** [reply]
https://github.com/k3s-io/k3s/pull/3231 ?

**Comment by jonstelly** [comment]
> There is not currently any tool to directly migrate between backends.

What about if we're using etcd but didn't do the cluster-init when we started, just spun up a single master cluster?  I just realized the other day when reading another issue that we can change the command-line arguments after install.

If I re-run the get.k3s.io script on my master and add --cluster-init, can I then add other masters?  Is that the correct/easiest way to do that?

**Reply by ChristianCiach** [reply]
How are you using etcd if you didn't install using `--cluster-init`? Are you using an external etcd cluster and now want to migrate to an embedded etcd?

**Reply by jonstelly** [reply]
Oh, I made an assumption that when etcd clustering was rolled out, embedded etcd became the default backend.  I guess it's still sqlite?  Thanks for pointing that out.

**Comment by ChristianCiach** [comment]
Looks like this is coming: https://github.com/k3s-io/k3s/issues/3982
